lib/datadog/appsec/response.rb

```diff
@@ -19,100 +19,38 @@ def to_rack
         [status, headers, body]
       end
 
-      def to_sinatra_response
-        ::Sinatra::Response.new(body, status, headers)
-      end
-
-      def to_action_dispatch_response
-        ::ActionDispatch::Response.new(status, headers, body)
-      end
-
       class << self
-        def negotiate(env, actions)
-          # @type var configured_response: Response?
-          configured_response = nil
-          actions.each do |type, parameters|
-            # Need to use next to make steep happy :(
-            # I rather use break to stop the execution
-            next if configured_response
-
-            configured_response = case type
-                                  when 'block_request'
-                                    block_response(env, parameters)
-                                  when 'redirect_request'
-                                    redirect_response(env, parameters)
-                                  end
-          end
-
-          configured_response || default_response(env)
-        end
-
-        def graphql_response(gateway_multiplex)
-          multiplex_return = []
-          gateway_multiplex.queries.each do |query|
-            # This method is only called in places where GraphQL-Ruby is already required
-            query_result = ::GraphQL::Query::Result.new(
-              query: query,
-              values: JSON.parse(content('application/json'))
-            )
-            multiplex_return << query_result
-          end
+        def build(action_params, http_accept_header)
```

Comment:
IMHO, we should signal with the name, that we can convert interrupt params into the response, hence, I would like to suggest naming like

1. `from_interrupt_params`
2. `from_interrupt`
3. `from_interrupt_signal`

or any other than `build` which is generic and doesn't reflect the fact that it will work solely with actions from the interrupt. In addition, do we want to convert accept headers into named kwarg?

```ruby
def from_interrupt_params(params, http_accept_header:)
```

Reply:
this is unfortunately not only for interruption, but also for stack trace generation and other actions. How about `from_actions`?

Reply:
sorry, misread your comment first, `from_interrupt_params` is better